src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1391,69 +1422,80 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxLis
                     this.ParseTypeParameterConstraintClauses(constraints);
                 }
 
-                var openBrace = this.EatToken(SyntaxKind.OpenBraceToken);
-
-                // ignore members if missing type name or missing open curly
-                if (name.IsMissing || openBrace.IsMissing)
+                SyntaxToken semicolon;
+                SyntaxToken? openBrace;
+                SyntaxToken? closeBrace;
+                if (isInterface || CurrentToken.Kind != SyntaxKind.SemicolonToken)
```

Comment:
>CurrentToken.Kind != SyntaxKind.SemicolonToken [](start = 35, length = 46)

I feel like this isn't a great solution. I know this change was prompted by Neal, but I feel like a better solution would be to parse the interface as a record declaration, and then issue a semantic error for "Interfaces cannot be records."